Add warning for code generation

kpcyrd opened the pull request:
This resolves #673. The first commit adds it to the template and the second commit are the generated changes. Please let me know if the comment wasn't final and should be changed to something else. :)

mthjones commented:
Thanks for adding that! There are a couple other files that get auto-generated that would be useful to add the header to as well: `lib.rs`, `Cargo.toml` and `README.md`. `lib.rs` should be easy enough (same as `generated.rs`). From some brief research, comments can be added to `toml` files with a `#` prefix and can be put in Markdown inside a `<!--- ... --->`.

mthjones commented on service_crategen/src/main.rs:
```diff
@@ -261,6 +261,18 @@ See [LICENSE][license] for details.
 //!
 //! If you're using the service, you're probably looking for [{client_name}](struct.{client_name}.html) and [{trait_name}](trait.{trait_name}.html).
 
+// =================================================================
```
I think the warning should be at the top of the file, even before the module doc comment. It doesn't interfere with doc generation.

matthewkmayer replied:
Agreed.  @kpcyrd can you move this to the top of this string so it's the first thing in the generated files?  Thanks!

kpcyrd commented:
Sorry for the delay, I've moved the comment, rebased and pushed the new commits :)

mthjones commented:
Hey @kpcyrd, thanks for following up!

It looks like the header got removed from `generated.rs` when it got added to `lib.rs`. Since both files are generated, it should be on both files. 😄

matthewkmayer commented:
What can I do to help finish this up? 😄

adimarco commented:
Hey @kpcyrd - you still interested in wrapping this up?  If not, I think someone else will likely grab it and finish it.  It's definitely something we want to get merged.

kpcyrd commented:
@adimarco sorry, I was running low on time. I figured out how to add the comment to generated.rs (I missed that file before the rebase as well).

I squashed and rebased on the latest master to resolve the merge conflict. :)

adimarco commented:
Looks good, and the generated docs look right too.  Thanks!
